pkg/providers/aws/tags.go

```diff
@@ -98,6 +100,33 @@ func genericToEc2Tags(tags []*tag) []*ec2.Tag {
 	return ec2Tags
 }
 
+// sort the keys to ensure tags returned in same order
+func mapToGeneric(tags map[string]string) []*tag {
+	var genTags []*tag
+	keys := make([]string, 0, len(tags))
+	for k := range tags {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Why the sort? does the order matter in CRO for a list of tags? I would not expect this list to be very large.

If you don't care about the sorting I thinking both these functions could be shortened to something like : 
```
	var dupMap = make(map[string]bool)
	for _, tag := range generalTags {
		dupMap[tag.key] = true
	}

	for _, tag := range infraTags {
		if _, exists := dupMap[tag.key]; !exists {
			generalTags = append(generalTags, tag)
		}
	}
	return generalTags
```

IMO I would try and avoid converting tags to a map[string]string and then converting them back to an array of tags. 

If you want to keep the sort, I would also suggest adding a sort to the generic tag type, can use this as an example - https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/providers/aws/sort.go

Reply:
The sort was just to ensure the order the tags are retrieved is consistent.

Your solution is a little more elegant so I definitely prefer this over converting to map and back again - made changes to use this instead.